Formula/m4.rb

```diff
@@ -30,6 +30,14 @@ class M4 < Formula
     end
   end
 
+  if OS.linux? && Hardware::CPU.arm?
+    # Apply fix for https://askubuntu.com/a/1112101
+    patch do
+      url "https://git.openembedded.org/openembedded-core/plain/meta/recipes-devtools/m4/m4/m4-1.4.18-glibc-change-work-around.patch"
+      sha256 "9e8fc660515be565f26194d3d6124a9483dc15a66ee8ea3f0b409c02d60a62f5"
+    end
+  end
```

Comment:
```suggestion
  on_linux do
    if Hardware::CPU.arm?
      # Apply fix for https://askubuntu.com/a/1112101
      patch do
        url "https://git.openembedded.org/openembedded-core/plain/meta/recipes-devtools/m4/m4/m4-1.4.18-glibc-change-work-around.patch"
        sha256 "9e8fc660515be565f26194d3d6124a9483dc15a66ee8ea3f0b409c02d60a62f5"
      end
    end
  end
```

Reply:
I think we're trying to stop using `if OS.linux?` and replacing them with `on_linux` blocks instead.